content/knowledge/flare_architecure.md

```diff
@@ -0,0 +1,207 @@
+
+Avalanche is considered to be a sort of layered consensus protocol and can be broken down into:
+
+* Slush
+* Snowflake
+* Snowball
+* Avalanche
+
+I'll keep this part brief and more detailed explanations can be found in the links provided at the end of this section.
+
+To understand how the algorithm works, here is a simple example.
+We start with a _slush_.
+Let's say, we have 16 nodes and the network has received a malicious transaction.
+Now these nodes should 'vote' whether this transaction is valid (the sender has sufficient funds in this case).
+Another assumption is that several of these nodes are malicious (I run them in the network to get my transaction through) or they are out-of-date (information they have regarding the state of the network is a bit obsolete).
+To achieve an agreement regarding the transaction, each node picks several other nodes randomly, and they have a vote.
+Each node can repeat this process several times to achieve a final agreement.
+However, after each voting, the node can change its preferences depending on the decision made by the majority.
+This can potentially make the process stuck at some point because the nodes can not achieve consensus.
+To break this loop, _snowflake_ comes into play.
+Each node has a counter to keep track of the votes.
+Each time the node votes with the same preference, counter increments.
+If the node changes its preference, the counter goes to zero and the node starts the process from the beginning.
+It will do so until the number of consequence votes reaches some defined threshold.
+This however can slow down the voting process, especially if the network has a big amount of malicious nodes.
+To speed this up, _snowball_ upgrades snowflake's counters to 'confidence' counters.
+Two counters hold information about the 'positive' and 'negative' votes.
+After some number of votes, the algorithm compares two counters and makes a decision based on that.
+This speeds up the voting process and dramatically decreases the probability of the voting process getting stuck.
+The final part is the Avalanche algorithm that generalizes the snowball and maintains the DAG (direct acyclic graph).
+DAG is a critical part because it prevents double-spends.
+Because in DAG the transactions are appended to one another, a vote for one transaction implicitly means a vote for all the transactions it is appended to.
+When the transaction is being created, it holds the information about one or more parent transactions.
+Each node then can check the transaction and all the ancestors and decide whether to accept or reject the transaction.
+In addition, the node queries each transaction only once.
+A final decision is made if the transaction is present in the consequent queries and has descendants (in case there are two conflicting transactions, one with more descendants will more likely to be accepted, because it will appear in more queries than another).
```

Comment:
Since the purpose of this document is to be a technical document, I'd suggest avoiding any form of story-telling examples, and to focus on a more factual/down-to-earth approach. It is also recommended to avoid using subjective terms like _a bit_, and to avoid using the future tense.

For example here, this could be changed to:

```suggestion
The following is an example of how the Avalanche Consensus Protocol behaves:

If a 16 nodes network receives a malicious transaction while using the Slush consensus protocol, its nodes 'vote' on whether this transaction is valid (the sender has sufficient funds in this case).
If several of these nodes are malicious or they are out-of-date (information they have regarding the state of the network is obsolete).
To achieve an agreement regarding the transaction, each node picks several other nodes randomly, and they have a vote <insert what the vote determines, here>.
Each node can repeat this process several times to achieve a final agreement.
However, after each voting, nodes can change their preferences depending on the decision made by the majority.
This can potentially halt the process if the nodes can not achieve consensus.

To solve this issue, _snowflake_ comes into play.
Each node has a counter to keep track of the votes.
Each time a node votes with the same preference, counter increments.
If the node changes its preference, the counter goes to zero and the node starts the process from the beginning.
It does so until the number of consequence votes reaches some defined threshold.
This, however, can slow down the voting process, especially if the network has a substantial amount of malicious nodes.

To speed this up, _snowball_ upgrades snowflake's counters to 'confidence' counters.
Two counters hold information about the 'positive' and 'negative' votes.
After some number of votes, the algorithm compares two counters and makes a decision based on that.
This speeds up the voting process and dramatically decreases the probability of the voting process getting stuck.

The final part is the Avalanche algorithm that generalizes the snowball and maintains the DAG (direct acyclic graph).
DAG is a critical part because it prevents double-spends.
Because in DAG the transactions are appended to one another, a vote for one transaction implicitly means a vote for all the transactions it is appended to.
When the transaction is being created, it holds the information about one or more parent transactions.
Each node then can check the transaction and all the ancestors and decide whether to accept or reject the transaction.
In addition, the node queries each transaction only once.
A final decision is made if the transaction is present in the consequent queries and has descendants (if there are two conflicting transactions, the one with more descendants is more likely to be accepted, because it appears in more queries than another).
```

What are consequence votes? This term is used but never explained. I would suggest either adding a link, or briefly explaining it. The same goes about the DAG, unless it's already in a glossary somewhere (which I think it might be) in which case we probably want to link to the glossary.